### PR TITLE
Add an extra to install ALL plugins

### PR DESCRIPTION
**Pull request by lukaspie**
Add an extra called `[all]` that installs ALL the reader plugins. We can use this in NOMAD so that we always have all of them available on the NOMAD main.

Also adds some explanations on plugins to the README.

There is the latest nexus-definitions commit in here, but this is very small, so I think it is fine to keep it here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,23 @@ data into the NeXus standard and visualising the files content.
...
 
+# Plugins
+There are a number of plugins available for pynxtools. These are extensions of pynxtools used for reading data of specialized experimental techniques.
+- [**pynxtools-mpes**](https://github.com/FAIRmat-NFDI/pynxtools-mpes): A reader for multi-dimensional photoelectron spectroscopy data.
+- [**pynxtools-stm**](https://github.com/FAIRmat-NFDI/pynxtools-stm): A reader for scanning tunneling microscopy (SPM) and spectroscopy (STS) data.
+- [**pynxtools-xps**](https://github.com/FAIRmat-NFDI/pynxtools-xps): A reader for X-ray photoelectron spectroscopy (XPS) data.
+
+You can install each of the plugins together with `pynxtools` by passing the name of the plugin as an extra to the pip install call. For example, for the `pynxtools-mpes` plugin:
+```shell
+pip install git+https://github.com/FAIRmat-NFDI/pynxtools.git#egg=pynxtools[mpes]
```
[the remark this answers is not in the document]
Comment (lukaspie) [reply README.md:59]: Yeah, I was planning to change it before we make a new release of the package.

Actually, let's change it now so we dont forget about it. It won't be too long until the next release.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,23 @@ data into the NeXus standard and visualising the files content.
...
+```
+
+In addition, you can also install ALL plugins with `pynxtools` by passing the `[all]` extra to the pip install call:
```
[the remark this answers is not in the document]
Comment (lukaspie) [reply README.md:62]: Yeah, makes sense, I'll change it.

File: README.md [hunk 3]
```diff
@@ -48,6 +48,23 @@ data into the NeXus standard and visualising the files content.
...
+```shell
+pip install git+https://github.com/FAIRmat-NFDI/pynxtools.git#egg=pynxtools[all]
```
[the remark this answers is not in the document]
Comment (lukaspie) [reply README.md:64]: changed

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (domna) [reply README.md:59]: Yes, let's also change it there

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukaspie) [reply README.md]: fixed

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukaspie) [reply README.md]: I added instruction to install the latest stable version from PyPI and kept the installation guide for the latest development version from github. Should be clear enough now.

**Comment by lukaspie** [comment]
Should be good to go now.

**Review by domna — APPROVED** [review]

**Pull request merged**